crates/pink-libs/subrpc/src/lib.rs

```diff
@@ -156,6 +143,49 @@ pub fn get_block_hash(
     Ok(H256(hash))
 }
 
+/// Gets revalant block header by given block hash
+/// Only work with chains define generic patterns like `Header<u32, BlakeTwo256>`
+pub fn get_header(
+    rpc_node: &str,
+    block_hash: Option<H256>,
+) -> core::result::Result<BlockHeaderOk, Error> {
+    let param = block_hash.map_or("null".to_string(), |h| format!("\"0x{h:x}\""));
+
+    let data =
+        format!(r#"{{"id":1, "jsonrpc":"2.0", "method": "chain_getHeader","params":[{param}]}}"#)
+            .into_bytes();
+    let resp_body = call_rpc(rpc_node, data)?;
+    let header: BlockHeader = json::from_slice(&resp_body).or(Err(Error::InvalidBody))?;
+    let header_result = header.result;
+
+    Ok(BlockHeaderOk {
+        parent_hash: header_result.parent_hash,
+        number: header_result.number,
+        state_root: header_result.state_root,
+        extrinsics_root: header_result.extrinsics_root,
+        digest: header_result.digest,
+    })
+}
+
+#[derive(Default, Encode, Decode, Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub struct ExtraParam {
```

Comment:
I suggest to add `Option<Era>`. If it's not specified, we fall back to `compute_era`.

Reply:
> This PR introduced back `sp-runtime`, which has a relatively large dependencies. I used to met problems when compiling ink contracts with `sp-runtime`. Also, we need to consider the wasm binary size increment.
> 
> In my understanding, only `Era` and `Digest` are used. In the past, `Era` was used. So I picked it out and copied it from `sp-runtime`. I checked `Digest`. It looks easier to copy than `Era`. So if possible, I'd suggest to copy it out again.

Let me try it

Reply:
> I suggest to add `Option<Era>`. If it's not specified, we fall back to `compute_era`.

I was considering if making fallback a default behavior is good,  seems there is no better solution. Let's do it